test/helpers/kubectl.go

```diff
@@ -4073,3 +4073,22 @@ func logGathererSelector(allNodes bool) string {
 
 	return selector
 }
+
+// GetDNSProxyPort returns the port the Cilium DNS proxy is listening on
+func (kub *Kubectl) GetDNSProxyPort(ciliumPod string) int {
+	const pickDNSProxyPort = `iptables-save | grep "udp.*TPROXY to host cilium-dns-egress" | sed 's/^.*--on-port \([1-9][0-9]*\).*$/\1/'`
+
+	// Find out the DNS proxy ports in use
+	res := kub.CiliumExecContext(context.TODO(), ciliumPod, pickDNSProxyPort)
+	if !res.WasSuccessful() {
+		ginkgoext.Failf("Cannot find DNS proxy port on %s", ciliumPod)
+		return 0
+	}
+	portStr := res.GetStdOut().String()
+	gomega.Expect(portStr).ShouldNot(gomega.BeEmpty(), "No DNS proxy port found on %s", ciliumPod)
```

Comment:
Please use `ExpectWithOffset`, this will make test failure point to parent call instead of this line.

Reply:
OK, so the offset value of 1 will point the failure to caller, 2 would point the failure to the caller's caller etc.?